scripts/zones/Northern_San_dOria/npcs/Ailbeche.lua

```diff
@@ -15,9 +15,9 @@ local entity = {}
 
 entity.onTrade = function(player, npc, trade)
     if (player:getCharVar("aBoysDreamCS") >= 3) then
-        if (trade:hasItemQty(17001, 1) == true and trade:getItemCount() == 1 and player:hasItem(4562) == false) then
+        if (trade:hasItemQty(17001, 1) == true and trade:getItemCount() == 1 and player:getCharVar("aBoysDreamCS") == 3) then
```

Comment:
We can now reduce the first two trade checks into:
```suggestion
        if npcUtil.tradeHasExactly(trade, xi.items.GIANT_SHELL_BUG) and player:getCharVar("aBoysDreamCS") == 3) then
```
You'll want to make sure this lua has the following library:
require("scripts/globals/npc_util")
You may have to add GIANT_SHELL_BUG to items.lua. Very easy to enter in, but absolutely necessary.
Same goes for below, but the item would be xi.items.ODONTOTYRANNUS

Reply:
I just modified what I knew.  I'll check this out in a bit.

Reply:
Look great! Thank you :) it's always good to fix these kinds things when we come by them.